etcdserver/v3demo_server.go

```diff
@@ -23,8 +23,52 @@ import (
 	dstorage "github.com/coreos/etcd/storage"
 )
 
-type V3DemoServer interface {
-	V3DemoDo(ctx context.Context, r pb.InternalRaftRequest) (proto.Message, error)
+type RaftKV interface {
```

Comment:
I will add docs when I feel good about the whole thing. I am still in the process of cleaning this up. The main goal of the pr is to separate things up.